src/MetadataUtility/Audio/Wamd.cs

```diff
@@ -0,0 +1,203 @@
+        /// </summary>
+        /// <param name="stream">The file stream.</param>
+        /// <returns>Boolean indicating weather the wamd chunk is present.</returns>
+        public static Fin<bool> HasVersion1WamdChunk(Stream stream)
+        {
+            var wamdChunk = GetWamdChunk(stream);
+
+            if (wamdChunk.IsFail)
+            {
+                return (Error)wamdChunk;
+            }
+
+            var wamdSpan = RangeHelper.ReadRange(stream, (RangeHelper.Range)wamdChunk);
+
+            var version = GetVersion(wamdSpan);
+
+            if (version.IsFail)
+            {
+                return (Error)version;
+            }
+
+            return version == 1;
+        }
+
+        /// <summary>
+        /// Scans for the position of the wamd chunk in the file stream.
+        /// </summary>
+        /// <param name="stream">The file stream.</param>
+        /// <returns>Range representing the position of the wamd chunk.</returns>
+        public static Fin<RangeHelper.Range> GetWamdChunk(Stream stream)
+        {
+            var riffChunk = Wave.FindRiffChunk(stream);
+            var waveChunk = riffChunk.Bind(r => Wave.FindWaveChunk(stream, r));
+            var wamdChunk = waveChunk.Bind(w => Wave.ScanForChunk(stream, w, WamdChunkId));
+
+            if (wamdChunk.IsFail)
+            {
+                return (Error)wamdChunk;
+            }
+
+            return wamdChunk;
+        }
+
+        /// <summary>
+        /// Extracts the wamd chunk version.
+        /// Wamd version has a sub chunk ID of 0, and is always first in the wamd chunk.
+        /// </summary>
+        /// <param name="wamdSpan">The wamd chunk.</param>
+        /// <returns>Wamd chunk version.</returns>
+        public static Fin<ushort> GetVersion(ReadOnlySpan<byte> wamdSpan)
+        {
+            int wamdOffset = 0;
+
+            ushort subChunkId = BinaryPrimitives.ReadUInt16LittleEndian(wamdSpan[wamdOffset..]);
+            wamdOffset += 2;
+
+            uint length = BinaryPrimitives.ReadUInt32LittleEndian(wamdSpan[wamdOffset..]);
+            wamdOffset += 4;
+
+            if (subChunkId != 0 || length != 2)
+            {
+                return WamdVersionError;
+            }
+
+            ushort version = BinaryPrimitives.ReadUInt16LittleEndian(wamdSpan[wamdOffset..]);
+
+            return version;
+        }
+
+        /// <summary>
+        /// Parses a date into Nodatime's OffsetDateTime.
+        /// </summary>
+        /// <param name="value">The date to parse.</param>
+        /// <returns>The parsed date.</returns>
+        public static OffsetDateTime? DateParser(string value)
+        {
+            OffsetDateTime? date = null;
+
+            try
+            {
+                date = OffsetDateTimePattern.CreateWithInvariantCulture("yyyy'-'MM'-'dd' 'HH':'mm':'sso<m>").Parse(value).Value;
+            }
+            catch (UnparsableValueException)
+            {
+                return null;
+            }
```

Comment:
A couple of notes here:

- we probably want EMU to crash if the string value is not-empty and we can't parse the date; this scenario would be a date format we don't understand and we want to halt, throw an exception, and hopefully get a user that reports the bug so we can improve emu.
- NodaTime specifically returns an type that can be checked for failure, so we can avoid exception-based logic, why is this wrapper even necesary?

On the assumption that the answer to the above is because it fails when  `value` is null or empty, I'd suggest checking that first, and returning null in those cases before attempting to do the parsing.

Reply:
Fixed this up but wanted to clarify. As of my most recent changes it should crash with an unrecognized date format. I could instead use some Fin logic to log the error and keep going?

Reply:
Unrecognized format? In the Wamd chunk?

Yeah an exception I think.

In a filename, we'd not crash, but that's because we're attempting parse dozens of possible formats for which there may not even be a date included! 

In wamd the behavior is clearly defined by a doc so any behavior outside of that is exceptional.